src/platform/packages/shared/kbn-unified-tabs/src/components/tab/tab.tsx

```diff
@@ -0,0 +1,158 @@
+  item: TabItem;
+  isSelected: boolean;
+  tabContentId: string;
+  onSelect: (item: TabItem) => void;
+  onClose: (item: TabItem) => void;
+}
+
+export const Tab: React.FC<TabProps> = ({ item, isSelected, tabContentId, onSelect, onClose }) => {
+  const { euiTheme } = useEuiTheme();
+
+  const tabContainerDataTestSubj = `unifiedTabs_tab_${item.id}`;
+  const closeButtonLabel = i18n.translate('unifiedTabs.closeTabButton', {
+    defaultMessage: 'Close',
+  });
+
+  const onSelectEvent = useCallback(
+    (event: MouseEvent<HTMLElement>) => {
+      event.stopPropagation();
+
+      if (!isSelected) {
+        onSelect(item);
+      }
+    },
+    [onSelect, item, isSelected]
+  );
+
+  const onCloseEvent = useCallback(
+    (event: MouseEvent<HTMLButtonElement>) => {
+      event.stopPropagation();
+      onClose(item);
+    },
+    [onClose, item]
+  );
+
+  const onClickEvent = useCallback(
+    (event: MouseEvent<HTMLDivElement>) => {
+      if (event.currentTarget.getAttribute('data-test-subj') === tabContainerDataTestSubj) {
+        // if user presses on the space around the buttons, we should still trigger the onSelectEvent
+        onSelectEvent(event);
+      }
+    },
+    [onSelectEvent, tabContainerDataTestSubj]
+  );
+
+  return (
+    <EuiFlexGroup
+      alignItems="center"
+      css={getTabContainerCss(euiTheme, isSelected)}
+      data-test-subj={tabContainerDataTestSubj}
+      responsive={false}
+      gutterSize="none"
+      onClick={onClickEvent}
+    >
+      <button
+        {...getTabAttributes(item, tabContentId)}
+        aria-selected={isSelected}
+        css={getTabButtonCss(euiTheme)}
+        className="unifiedTabs__tabBtn"
+        data-test-subj={`unifiedTabs_selectTabBtn_${item.id}`}
+        role="tab"
+        type="button"
+        onClick={onSelectEvent}
+      >
+        <EuiText color="inherit" size="s" className="eui-textTruncate">
+          {item.label}
+        </EuiText>
+      </button>
+      <EuiFlexItem grow={false} className="unifiedTabs__closeTabBtn">
+        <EuiButtonIcon
+          aria-label={closeButtonLabel}
+          title={closeButtonLabel}
+          color="text"
+          data-test-subj={`unifiedTabs_closeTabBtn_${item.id}`}
+          iconType="cross"
+          onClick={onCloseEvent}
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+};
+
+function getTabContainerCss(euiTheme: EuiThemeComputed, isSelected: boolean) {
+  // TODO: remove the usage of deprecated colors
+
+  return css`
+    display: inline-flex;
+    border-right: ${euiTheme.border.thin};
+    height: ${euiTheme.size.xl};
+    padding-left: ${euiTheme.size.m};
+    padding-right: ${euiTheme.size.xs};
+    min-width: 96px;
+    max-width: 280px;
+
+    background-color: ${isSelected ? euiTheme.colors.emptyShade : euiTheme.colors.lightestShade};
+    color: ${isSelected ? euiTheme.colors.text : euiTheme.colors.subduedText};
+    transition: background-color ${euiTheme.animation.normal};
```

Comment:
I'm not sure if there were specs for this or not, but from testing locally I found the animations a bit slow and it felt sort of laggy. If we add animations for this, personally I'd go with something like `extraFast` or `fast`. Maybe @l-suarez has thoughts on this one too?